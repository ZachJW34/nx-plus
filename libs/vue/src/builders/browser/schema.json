{
  "$id": "https://json-schema.org/draft-07/schema",
  "title": "Browser builder",
  "description": "Build app",
  "type": "object",
  "properties": {
    "mode": {
      "type": "string",
      "description": "Specify env mode (default: development).",
      "default": "development"
    },
    "dest": {
      "type": "string",
      "description": "Specify output directory."
    },
    "clean": {
      "type": "boolean",
      "description": "Remove the dist directory before building the project.",
      "default": true
    },
    "report": {
      "type": "boolean",
      "description": "Generate report.html to help analyze bundle content.",
      "default": false
    },
    "reportJson": {
      "type": "boolean",
      "description": "Generate report.json to help analyze bundle content.",
      "default": false
    },
    "skipPlugins": {
      "type": "string",
      "description": "Comma-separated list of plugin names to skip for this run."
    },
    "watch": {
      "type": "boolean",
      "description": "Watch for changes.",
      "default": false
    },
    "index": {
      "type": "string",
      "description": "The path of a file to use for the application's HTML index. The filename of the specified path will be used for the generated file and will be created in the root of the application's configured output path."
    },
    "main": {
      "type": "string",
      "description": "The full path for the main entry point to the app, relative to the current workspace."
    },
    "tsConfig": {
      "type": "string",
      "description": "The full path for the TypeScript configuration file, relative to the current workspace."
    },
    "publicPath": {
      "type": "string",
      "description": "The base URL your application bundle will be deployed at.",
      "default": "/"
    },
    "filenameHashing": {
      "type": "boolean",
      "description": "Generated static assets contains hashes in their filenames for better caching control.",
      "default": false
    },
    "productionSourceMap": {
      "type": "boolean",
      "description": "Setting this to false can speed up production builds if you don't need source maps for production.",
      "default": false
    },
    "css": {
      "type": "object",
      "properties": {
        "requireModuleExtension": {
          "type": "boolean",
          "description": "By default, only files that ends in *.module.[ext] are treated as CSS modules. Setting this to false will allow you to drop .module in the filenames and treat all *.(css|scss|sass|less|styl(us)?) files as CSS modules.",
          "default": true
        },
        "extract": {
          "description": "Whether to extract CSS in your components into a standalone CSS files (instead of inlined in JavaScript and injected dynamically).",
          "oneOf": [
            {
              "type": "boolean"
            },
            {
              "type": "object"
            }
          ],
          "default": false
        },
        "sourceMap": {
          "type": "boolean",
          "description": "Whether to enable source maps for CSS. Setting this to true may affect build performance.",
          "default": false
        },
        "loaderOptions": {
          "type": "object",
          "description": "Pass options to CSS-related loaders.",
          "default": {}
        }
      },
      "additionalProperties": false
    },
    "stdin": {
      "type": "boolean",
      "description": "Close when stdin ends.",
      "default": false
    }
  },
  "required": ["dest", "index", "main", "tsConfig"],
  "additionalProperties": false
}
