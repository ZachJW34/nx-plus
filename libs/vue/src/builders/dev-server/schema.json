{
  "$id": "https://json-schema.org/draft-07/schema",
  "title": "DevServer builder",
  "description": "Start development server",
  "type": "object",
  "properties": {
    "open": {
      "type": "boolean",
      "description": "Open browser on server start.",
      "default": false
    },
    "copy": {
      "type": "boolean",
      "description": "Copy url to clipboard on server start.",
      "default": false
    },
    "stdin": {
      "type": "boolean",
      "description": "Close when stdin ends."
    },
    "mode": {
      "type": "string",
      "description": "Specify env mode (default: development)."
    },
    "host": {
      "type": "string",
      "description": "Specify host (default: 0.0.0.0).",
      "default": "0.0.0.0"
    },
    "port": {
      "type": "number",
      "description": "Specify port (default: 8080).",
      "default": 8080
    },
    "https": {
      "type": "boolean",
      "description": "Use https (default: false).",
      "default": false
    },
    "public": {
      "type": "string",
      "description": "Specify the public network URL for the HMR client."
    },
    "skipPlugins": {
      "type": "string",
      "description": "Comma-separated list of plugin names to skip for this run."
    },
    "browserTarget": {
      "type": "string",
      "description": "Target to serve.",
      "pattern": "^[^:\\s]+:[^:\\s]+(:[^\\s]+)?$"
    },
    "watch": {
      "type": "boolean",
      "description": "Watch for changes.",
      "default": true
    },
    "publicPath": {
      "type": "string",
      "description": "The base URL your application bundle will be deployed at."
    },
    "css": {
      "type": "object",
      "properties": {
        "requireModuleExtension": {
          "type": "boolean",
          "description": "By default, only files that ends in *.module.[ext] are treated as CSS modules. Setting this to false will allow you to drop .module in the filenames and treat all *.(css|scss|sass|less|styl(us)?) files as CSS modules."
        },
        "extract": {
          "description": "Whether to extract CSS in your components into a standalone CSS files (instead of inlined in JavaScript and injected dynamically).",
          "oneOf": [
            {
              "type": "boolean"
            },
            {
              "type": "object"
            }
          ]
        },
        "sourceMap": {
          "type": "boolean",
          "description": "Whether to enable source maps for CSS. Setting this to true may affect build performance."
        },
        "loaderOptions": {
          "type": "object",
          "description": "Pass options to CSS-related loaders."
        }
      },
      "additionalProperties": false
    },
    "devServer": {
      "type": "object",
      "description": "All options for webpack-dev-server are supported.",
      "default": {}
    }
  },
  "required": ["browserTarget"],
  "additionalProperties": false
}
