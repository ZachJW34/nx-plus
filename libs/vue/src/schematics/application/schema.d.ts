export interface ApplicationSchematicSchema {
  name: string;
  tags?: string;
  directory?: string;
  style: 'css' | 'scss' | 'less' | 'stylus';
  unitTestRunner: 'jest' | 'none';
  e2eTestRunner: 'cypress' | 'none';
  routing: boolean;
  vueVersion: number;
  skipFormat: boolean;
  babel: boolean;
}
