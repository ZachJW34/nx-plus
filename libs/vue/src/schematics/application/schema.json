{
  "$schema": "http://json-schema.org/schema",
  "$id": "Application",
  "title": "Vue Application Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the application",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "x-prompt": {
        "message": "Which style format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SCSS   [ https://sass-lang.com/documentation/syntax#scss ]"
          },
          {
            "value": "stylus",
            "label": "Stylus [ https://stylus-lang.com                         ]"
          },
          {
            "value": "less",
            "label": "LESS   [ http://lesscss.org                              ]"
          }
        ]
      }
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["jest", "none"],
      "description": "Test runner to use for unit tests",
      "default": "jest"
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["cypress", "none"],
      "description": "Test runner to use for end to end (e2e) tests",
      "default": "cypress"
    },
    "routing": {
      "type": "boolean",
      "description": "Generate routing configuration",
      "x-prompt": "Would you like to configure routing for this application?",
      "default": false
    },
    "vueVersion": {
      "description": "The version of Vue.js that you want to start the project with.",
      "type": "number",
      "default": 0,
      "x-prompt": {
        "message": "Which version of Vue.js would you like to use?",
        "type": "list",
        "items": [
          {
            "value": 2,
            "label": "2.x"
          },
          {
            "value": 3,
            "label": "3.x"
          }
        ]
      }
    },
    "skipFormat": {
      "type": "boolean",
      "description": "Skip formatting files",
      "default": false
    },
    "babel": {
      "type": "boolean",
      "description": "Add Babel support.",
      "default": false,
      "x-prompt": "Use Babel alongside TypeScript?"
    }
  },
  "required": ["name"]
}
