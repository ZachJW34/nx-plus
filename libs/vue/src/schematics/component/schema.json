{
  "$schema": "http://json-schema.org/schema",
  "$id": "Component",
  "title": "Create a Vue Component",
  "type": "object",
  "examples": [
    {
      "command": "g component my-component --project my-project",
      "description": "Generate a component in the my-project app"
    }
  ],
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of your component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the component?"
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this component?"
    },
    "directory": {
      "type": "string",
      "description": "Create the component under this directory (can be nested).",
      "alias": "d"
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "alias": "s",
      "default": "css",
      "x-prompt": {
        "message": "Which style format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SCSS   [ https://sass-lang.com/documentation/syntax#scss ]"
          },
          {
            "value": "stylus",
            "label": "Stylus [ https://stylus-lang.com                         ]"
          },
          {
            "value": "less",
            "label": "LESS   [ http://lesscss.org                              ]"
          }
        ]
      }
    }
  },
  "required": ["name", "project"]
}
