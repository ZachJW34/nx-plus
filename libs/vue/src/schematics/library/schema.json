{
  "$schema": "http://json-schema.org/schema",
  "$id": "Vue library",
  "title": "Create a Vue Library",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Library name",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the library?"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the lib is placed"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["jest", "none"],
      "description": "Test runner to use for unit tests",
      "default": "jest"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the library (used for linting)"
    },
    "skipFormat": {
      "description": "Skip formatting files",
      "type": "boolean",
      "default": false
    },
    "skipTsConfig": {
      "type": "boolean",
      "default": false,
      "description": "Do not update tsconfig.json for development experience."
    },
    "publishable": {
      "type": "boolean",
      "default": false,
      "description": "Create a buildable library."
    },
    "vueVersion": {
      "description": "The version of Vue.js that you want to start the project with.",
      "type": "number",
      "default": 0,
      "x-prompt": {
        "message": "Which version of Vue.js would you like to use?",
        "type": "list",
        "items": [
          {
            "value": 2,
            "label": "2.x"
          },
          {
            "value": 3,
            "label": "3.x"
          }
        ]
      }
    },
    "babel": {
      "type": "boolean",
      "description": "Add Babel support.",
      "default": false,
      "x-prompt": "Use Babel alongside TypeScript?"
    }
  },
  "required": ["name"],
  "additionalProperties": false
}
