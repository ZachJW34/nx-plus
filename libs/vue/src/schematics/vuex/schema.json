{
  "$schema": "http://json-schema.org/schema",
  "$id": "Vuex",
  "title": "Vuex Options Schema",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project you would like to add Vuex to",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What project would you like to add Vuex to?"
    },
    "skipFormat": {
      "type": "boolean",
      "description": "Skip formatting files",
      "default": false
    }
  },
  "required": ["project"]
}
