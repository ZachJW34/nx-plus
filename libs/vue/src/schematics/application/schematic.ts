import { tags } from '@angular-devkit/core';
import {
  apply,
  applyTemplates,
  chain,
  externalSchematic,
  filter,
  mergeWith,
  move,
  noop,
  Rule,
  Tree,
  url,
} from '@angular-devkit/schematics';
import {
  addDepsToPackageJson,
  addLintFiles,
  addPackageWithInit,
  addProjectToNxJsonInTree,
  formatFiles,
  generateProjectLint,
  Linter,
  names,
  offsetFromRoot,
  projectRootDir,
  ProjectType,
  toFileName,
  updateJsonInTree,
  updateWorkspace,
} from '@nrwl/workspace';
import { ApplicationSchematicSchema } from './schema';

/**
 * Depending on your needs, you can change this to either `Library` or `Application`
 */
const projectType = ProjectType.Application;

interface NormalizedSchema extends ApplicationSchematicSchema {
  projectName: string;
  projectRoot: string;
  projectDirectory: string;
  parsedTags: string[];
}

function normalizeOptions(
  options: ApplicationSchematicSchema
): NormalizedSchema {
  const name = toFileName(options.name);
  const projectDirectory = options.directory
    ? `${toFileName(options.directory)}/${name}`
    : name;
  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
  const projectRoot = `${projectRootDir(projectType)}/${projectDirectory}`;
  const parsedTags = options.tags
    ? options.tags.split(',').map((s) => s.trim())
    : [];

  return {
    ...options,
    projectName,
    projectRoot,
    projectDirectory,
    parsedTags,
  };
}

function addFiles(options: NormalizedSchema): Rule {
  return mergeWith(
    apply(url(`./files`), [
      applyTemplates({
        ...options,
        ...names(options.name),
        offsetFromRoot: offsetFromRoot(options.projectRoot),
        dot: '.',
        baseUrl: '<%= BASE_URL %>',
        htmlWebpackPluginTitle: '<%= htmlWebpackPlugin.options.title %>',
      }),
      options.unitTestRunner === 'none'
        ? filter((file) => file !== '/tests/unit/example.spec.ts')
        : noop(),
      options.routing
        ? noop()
        : filter(
            (file) =>
              ![
                '/src/router/index.ts',
                '/src/views/About.vue',
                '/src/views/Home.vue',
              ].includes(file)
          ),
      move(options.projectRoot),
    ])
  );
}

function getEslintConfig(options: NormalizedSchema) {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const eslintConfig: any = {
    extends: [
      'plugin:vue/essential',
      '@vue/typescript/recommended',
      'prettier',
      'prettier/@typescript-eslint',
    ],
    rules: {},
    env: {
      node: true,
    },
  };

  if (options.unitTestRunner === 'jest') {
    eslintConfig.overrides = [
      {
        files: ['**/*.spec.{j,t}s?(x)'],
        env: {
          jest: true,
        },
      },
    ];
  }

  return eslintConfig;
}

function addEsLint(options: NormalizedSchema): Rule {
  return chain([
    updateWorkspace((workspace) => {
      const { targets } = workspace.projects.get(options.projectName);
      targets.add({
        name: 'lint',
        ...generateProjectLint(
          options.projectRoot,
          `${options.projectRoot}/tsconfig.app.json`,
          Linter.EsLint
        ),
      });
    }),
    addLintFiles(options.projectRoot, Linter.EsLint, {
      localConfig: getEslintConfig(options),
    }),
    // Extending the root ESLint config should be the first value in the
    // app's local ESLint config extends array.
    updateJsonInTree(`${options.projectRoot}/.eslintrc`, (json) => {
      json.extends.unshift(json.extends.pop());
      return json;
    }),
    (tree: Tree) => {
      const configPath = `${options.projectRoot}/.eslintrc`;
      const content = tree.read(configPath).toString('utf-8').trim();
      tree.rename(configPath, `${configPath}.js`);
      tree.overwrite(`${configPath}.js`, `module.exports = ${content};`);
    },
  ]);
}

function addJest(options: NormalizedSchema): Rule {
  return chain([
    addPackageWithInit('@nrwl/jest'),
    externalSchematic('@nrwl/jest', 'jest-project', {
      project: options.projectName,
      setupFile: 'none',
      skipSerializers: true,
      supportTsx: true,
      testEnvironment: 'jsdom',
      babelJest: false,
    }),
    updateJsonInTree(`${options.projectRoot}/tsconfig.spec.json`, (json) => {
      json.include = json.include.filter((pattern) => !/\.jsx?$/.test(pattern));
      json.compilerOptions = {
        ...json.compilerOptions,
        jsx: 'preserve',
        esModuleInterop: true,
        allowSyntheticDefaultImports: true,
      };
      return json;
    }),
    (tree: Tree) => {
      const content = tags.stripIndent`
        module.exports = {
          name: '${options.projectName}',
          preset: '${offsetFromRoot(options.projectRoot)}jest.config.js',
          transform: {
            '^.+\\.vue$': 'vue-jest',
            '.+\\.(css|styl|less|sass|scss|svg|png|jpg|ttf|woff|woff2)$':
              'jest-transform-stub',
            '^.+\\.tsx?$': 'ts-jest'
          },
          moduleFileExtensions: ["ts", "tsx", "vue", "js", "json"],
          coverageDirectory: '${offsetFromRoot(options.projectRoot)}coverage/${
        options.projectRoot
      }',
          snapshotSerializers: ['jest-serializer-vue'],
          globals: { 'ts-jest': { tsConfig: '<rootDir>/tsconfig.spec.json' }, 'vue-jest': { tsConfig: '${
            options.projectRoot
          }/tsconfig.spec.json' } },
        };
      `;
      tree.overwrite(`${options.projectRoot}/jest.config.js`, content);
      return tree;
    },
    addDepsToPackageJson(
      {},
      {
        '@vue/test-utils': '^1.0.3',
        'babel-core': '^7.0.0-bridge.0',
        'jest-serializer-vue': '^2.0.2',
        'jest-transform-stub': '^2.0.0',
        'vue-jest': '^3.0.5',
      },
      true
    ),
  ]);
}

function addCypress(options: NormalizedSchema): Rule {
  return chain([
    addPackageWithInit('@nrwl/cypress'),
    externalSchematic('@nrwl/cypress', 'cypress-project', {
      project: options.projectName,
      name: options.name + '-e2e',
      directory: options.directory,
      linter: Linter.EsLint,
      js: false,
    }),
    (tree) => {
      const appSpecPath =
        options.projectRoot + '-e2e/src/integration/app.spec.ts';
      tree.overwrite(
        appSpecPath,
        tree
          .read(appSpecPath)
          .toString('utf-8')
          .replace(
            `Welcome to ${options.projectName}!`,
            'Welcome to Your Vue.js + TypeScript App'
          )
      );
    },
  ]);
}

function addPostInstall() {
  return updateJsonInTree('package.json', (json, context) => {
    const vuePostInstall =
      'node node_modules/@nx-plus/vue/patch-nx-dep-graph.js';
    const { postinstall } = json.scripts || {};
    if (postinstall) {
      if (postinstall !== vuePostInstall) {
        context.logger.warn(
          "We couldn't add our postinstall script. Without it Nx's dependency graph won't support Vue files. For more information see https://github.com/ZachJW34/nx-plus/tree/master/libs/vue#nx-dependency-graph-support"
        );
      }
      return json;
    }
    json.scripts = { ...json.scripts, postinstall: vuePostInstall };
    return json;
  });
}

export default function (options: ApplicationSchematicSchema): Rule {
  const normalizedOptions = normalizeOptions(options);
  return chain([
    updateWorkspace((workspace) => {
      const { targets } = workspace.projects.add({
        name: normalizedOptions.projectName,
        root: normalizedOptions.projectRoot,
        sourceRoot: `${normalizedOptions.projectRoot}/src`,
        projectType,
      });
      targets.add({
        name: 'build',
        builder: '@nx-plus/vue:browser',
        options: {
          dest: `dist/${normalizedOptions.projectRoot}`,
          index: `${normalizedOptions.projectRoot}/public/index.html`,
          main: `${normalizedOptions.projectRoot}/src/main.ts`,
          tsConfig: `${normalizedOptions.projectRoot}/tsconfig.app.json`,
        },
        configurations: {
          production: {
            mode: 'production',
            filenameHashing: true,
            productionSourceMap: true,
            css: {
              extract: true,
              sourceMap: false,
            },
          },
        },
      });
      targets.add({
        name: 'serve',
        builder: '@nx-plus/vue:dev-server',
        options: {
          browserTarget: `${normalizedOptions.projectName}:build`,
        },
        configurations: {
          production: {
            browserTarget: `${normalizedOptions.projectName}:build:production`,
          },
        },
      });
    }),
    addProjectToNxJsonInTree(normalizedOptions.projectName, {
      tags: normalizedOptions.parsedTags,
    }),
    addFiles(normalizedOptions),
    addEsLint(normalizedOptions),
    options.unitTestRunner === 'jest' ? addJest(normalizedOptions) : noop(),
    options.e2eTestRunner === 'cypress'
      ? addCypress(normalizedOptions)
      : noop(),
    addPostInstall(),
    addDepsToPackageJson(
      {
        vue: '^2.6.11',
        ...(options.routing ? { 'vue-router': '^3.2.0' } : {}),
      },
      {
        '@vue/cli-plugin-typescript': '~4.5.0',
        '@vue/cli-service': '~4.5.0',
        '@vue/eslint-config-typescript': '^5.0.2',
        'eslint-plugin-vue': '^6.2.2',
        'vue-template-compiler': '^2.6.11',
      },
      true
    ),
    formatFiles(options),
  ]);
}
