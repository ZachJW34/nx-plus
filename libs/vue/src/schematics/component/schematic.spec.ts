import { Tree } from '@angular-devkit/schematics';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { createEmptyWorkspace } from '@nrwl/workspace/testing';
import { join } from 'path';

import { ComponentSchematicSchema, Style } from './schema';
import { readJsonInTree, toClassName } from '@nrwl/workspace';
import { ApplicationSchematicSchema } from '../application/schema';

describe('component schematic', () => {
  let appTree: Tree;

  describe('for app', () => {
    beforeEach(async () => {
      appTree = createEmptyWorkspace(Tree.empty());

      const appOptions: ApplicationSchematicSchema = {
        name: 'my-app',
        unitTestRunner: 'jest',
        e2eTestRunner: 'cypress',
        routing: false,
        style: 'css',
        vueVersion: 3,
        skipFormat: false,
        babel: false,
      };
      const appTestRunner = new SchematicTestRunner(
        '@nx-plus/vue',
        join(__dirname, '../../../collection.json')
      );

      await appTestRunner
        .runSchematicAsync('application', appOptions, appTree)
        .toPromise();
    });

    describe('styles', () => {
      let options: ComponentSchematicSchema = {
        name: 'my-component',
        project: 'my-app',
        style: 'css',
      };

      const componentTestRunner = new SchematicTestRunner(
        '@nx-plus/vue',
        join(__dirname, '../../../collection.json')
      );

      it.each(['scss' as Style, 'styl' as Style, 'less' as Style])(
        'should set correct style %s',
        async (style) => {
          options = {
            ...options,
            style,
          };

          const tree = await componentTestRunner
            .runSchematicAsync('component', options, appTree)
            .toPromise();

          const workspaceJson = readJsonInTree(tree, 'workspace.json');
          const projectRoot = workspaceJson.projects['my-app'].sourceRoot;

          const componentPath = `${projectRoot}/${toClassName(
            options.name
          )}.vue`;

          const fileEntry = tree.get(componentPath).content.toString();

          expect(fileEntry).toContain(`lang="${style}"`);
        }
      );
    });

    describe('directory', () => {
      const options: ComponentSchematicSchema = {
        name: 'my-component',
        project: 'my-app',
        directory: 'ui',
        style: 'css',
      };

      const componentTestRunner = new SchematicTestRunner(
        '@nx-plus/vue',
        join(__dirname, '../../../collection.json')
      );

      it('should set directory when used', async () => {
        const tree = await componentTestRunner
          .runSchematicAsync('component', options, appTree)
          .toPromise();

        const workspaceJson = readJsonInTree(tree, 'workspace.json');
        const projectRoot = workspaceJson.projects['my-app'].sourceRoot;

        const componentPath = `${projectRoot}/ui/${toClassName(
          options.name
        )}.vue`;

        const fileEntry = tree.get(componentPath).content.toString();

        expect(fileEntry).toMatchSnapshot();
      });
    });
  });

  describe('for library', () => {
    beforeEach(async () => {
      appTree = createEmptyWorkspace(Tree.empty());

      const libOptions = {
        name: 'my-lib',
        unitTestRunner: 'jest',
        skipFormat: false,
        publishable: false,
        vueVersion: 3,
        skipTsConfig: false,
      };
      const libTestRunner = new SchematicTestRunner(
        '@nx-plus/vue',
        join(__dirname, '../../../collection.json')
      );

      await libTestRunner
        .runSchematicAsync('lib', libOptions, appTree)
        .toPromise();
    });

    describe('styles', () => {
      let options: ComponentSchematicSchema = {
        name: 'my-component',
        project: 'my-lib',
        style: 'css',
      };

      const componentTestRunner = new SchematicTestRunner(
        '@nx-plus/vue',
        join(__dirname, '../../../collection.json')
      );

      it.each(['scss' as Style, 'styl' as Style, 'less' as Style])(
        'should set correct style %s',
        async (style) => {
          options = {
            ...options,
            style,
          };

          const tree = await componentTestRunner
            .runSchematicAsync('component', options, appTree)
            .toPromise();

          const workspaceJson = readJsonInTree(tree, 'workspace.json');
          const projectRoot = workspaceJson.projects['my-lib'].sourceRoot;

          const componentPath = `${projectRoot}/lib/${toClassName(
            options.name
          )}.vue`;

          const fileEntry = tree.get(componentPath).content.toString();

          expect(fileEntry).toContain(`lang="${style}"`);
        }
      );
    });

    describe('directory', () => {
      const options: ComponentSchematicSchema = {
        name: 'my-component',
        project: 'my-lib',
        directory: 'components',
        style: 'css',
      };

      const componentTestRunner = new SchematicTestRunner(
        '@nx-plus/vue',
        join(__dirname, '../../../collection.json')
      );

      it('should set directory when used', async () => {
        const tree = await componentTestRunner
          .runSchematicAsync('component', options, appTree)
          .toPromise();

        const workspaceJson = readJsonInTree(tree, 'workspace.json');
        const projectRoot = workspaceJson.projects['my-lib'].sourceRoot;

        const componentPath = `${projectRoot}/lib/components/${toClassName(
          options.name
        )}.vue`;

        expect(tree.exists(componentPath)).toBeTruthy();
      });
    });
  });
});
