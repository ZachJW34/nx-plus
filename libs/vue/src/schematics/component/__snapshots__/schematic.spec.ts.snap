// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component schematic for app directory should set directory when used 1`] = `
"<template>
  <p>Welcome to MyComponent</p>
</template>

<script lang=\\"ts\\">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'MyComponent',
});
</script>

<style scoped></style>
"
`;
